runtime/lua/vim/treesitter/query.lua

```diff
@@ -165,11 +163,21 @@ function M.get_node_text(node, source)
   local end_row, end_col, end_byte = node:end_()
 
   if type(source) == "number" then
-    if start_row ~= end_row then
-      return nil
+    local lines = a.nvim_buf_get_lines(source, start_row, end_row+1, true)
+    lines[1] = string.sub(lines[1], start_col+1)
+
+    local end_index = end_col
+    if #lines == 1 then
+      end_index = end_col - start_col
     end
-    local line = a.nvim_buf_get_lines(source, start_row, start_row+1, true)[1]
-    return string.sub(line, start_col+1, end_col)
+    lines[#lines] = string.sub(lines[#lines], 1, end_index)
+
+    local text = lines[1]
+    for i=2,#lines do
+      text = text .. '\n' .. lines[i]
+    end
```

Comment:
table.concat?

Reply:
Changed, thanks

Reply:
Can you also check whether match? keeps working with Multi-Line input? I guess vim regexes are only single line so hopefully it doesn't errror

Reply:
#match? doesn't use get_node_text so this PR shouldn't break it. I'll be probably making another PR regarding #match?, I would like it to support multi-line nodes, to make it more consistent with the official tree-sitter rust bindings.

Reply:
Lua-match? Supports Multi-Line regexes and I guess it already uses get_nod_text

Reply:
lua-match does use get_node_text, but has it's own check that prevents it from working on multi-line nodes. Still works on single-line nodes.

```lua
      local start_row, _, end_row, _ = node:range()
      if start_row ~= end_row then
        return false
      end
```

Reply:
After removing the check it still works the same on single-line nodes, and works correctly on multi-line nodes. Should I do commit it in this PR, or make a new PR?

Reply:
It'll be good to have kn this PR as the changes are related

Reply:
commited https://github.com/neovim/neovim/pull/14445/commits/2b02772698b49ea8fb319f9762d38a9b4fde6211